modules/packages/freebsd_ports

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh -e
+
+command=$1
+
+while read -r line; do
+  export INPUT_$line
+done
+
+list_updates() {
+  pkg version -v -l "<" | sed -Ee $'s/([^ ]+)-[^-]+ .* (.*)\\)/Name=\\1\\\nVersion=\\2\\\nArchitecture=none/'
+}
+
+case $command in
+  supports-api-version)
+    echo 1
+  ;;
+  get-package-data)
+    name="${INPUT_File?File must be given to get-package-data}"
+    echo PackageType=repo
+    echo Name=$name
+  ;;
+  list-installed)
+    pkg info -f -a | grep -E '^(Name|Version|Architecture)' | sed -Ee 's/[ ]+:[ ]+/=/' -e 's/^Architecture=.*/Architecture=none/'
+  ;;
+  repo-install)
+    name="${INPUT_Name?Name must be given to repo-install}"
+    version="${INPUT_Version}"
+
+    export BATCH=1
+    cd `whereis -sq "$name"`
```

Comment:
It should handle it gracefully. ``get-package-data`` should be cheap, and should do as little checking as possible, so it's better to check it in ``repo-install``.